content/docs/rendering-elements.md

```diff
@@ -1,75 +1,77 @@
 
-It displays "Hello, world" on the page.
+## به‌روز‌رسانی المان رندر شده {#updating-the-rendered-element}
 
-## Updating the Rendered Element {#updating-the-rendered-element}
+المان‌های ری‌اکت [تغییر ناپدیر](https://en.wikipedia.org/wiki/Immutable_object) هستند. زمانی که یک المان ری‌اکت را ایجاد می‌کنید، دیگر قادر به تغییر فرزندان و خصوصیات آن نخواهید بود. یک المان را مانند یک فریم از یک فیلم تصور کنید که UI را در یک نقطه زمانی مشخص نشان می‌دهد.
 
-React elements are [immutable](https://en.wikipedia.org/wiki/Immutable_object). Once you create an element, you can't change its children or attributes. An element is like a single frame in a movie: it represents the UI at a certain point in time.
+با اطلاعاتی که تا این‌جا به‌دست آورده‌ایم، تنها راه به‌روز رسانی UI این است که یک المان جدید ساخته و آن را به تابع `ReactDOM.render()` بدهیم.
 
-With our knowledge so far, the only way to update the UI is to create a new element, and pass it to `ReactDOM.render()`.
 
-Consider this ticking clock example:
+این مثال ساعت را درنظر بگیرید:
 
 `embed:rendering-elements/update-rendered-element.js`
 
-[](codepen://rendering-elements/update-rendered-element)
+[با CodePen امتحان کنید](codepen://rendering-elements/update-rendered-element)
 
-It calls `ReactDOM.render()` every second from a [`setInterval()`](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) callback.
+در این مثال تابع `ReactDOM.render()` هربار توسط callback تابع [`setInterval()`](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) فراخوانی می‌شود.
 
->**Note:**
+>**یادداشت:**
 >
->In practice, most React apps only call `ReactDOM.render()` once. In the next sections we will learn how such code gets encapsulated into [stateful components](/docs/state-and-lifecycle.html).
+> در عمل، بیشتر برنامه‌های ری‌اکت فقط یک دفعه `ReactDOM.render()` را فراخوانی می‌کنند. در بخش‌های بعدی یاد می‌گیریم که چگونه چنین کد‌هایی به صورت [کامپوننت‌های دارای state](/docs/state-and-lifecycle.html) کپسوله می‌شوند.
 >
->We recommend that you don't skip topics because they build on each other.
+> از آن‌جا که هر بخش با استناد به موضوع‌های پیشین نوشته‌ شده‌است، پیشنهاد می‌کنیم به‌ترتیب پیش بروید.
 
-## React Only Updates What's Necessary {#react-only-updates-whats-necessary}
+## ری‌اکت تنها چیز‌هایی که نیاز‌ هست را به‌روز رسانی می‌کند {#react-only-updates-whats-necessary}
 
-React DOM compares the element and its children to the previous one, and only applies the DOM updates necessary to bring the DOM to the desired state.
+DOM ری‌اکت المان موردنظر و فرزندانش را با حالت پیشین آن مقایسه می‌کند و تنها به‌روز رسانی‌های مورد نیاز را اعمال می‌کند تا ‌DOM را به وضعیت موردنظر برساند.
 
-You can verify by inspecting the [last example](codepen://rendering-elements/update-rendered-element) with the browser tools:
+برای اطمینان، [مثال قبلی](codepen://rendering-elements/update-rendered-element) را با استفاده از ابزار توسعه‌دهندگان مرورگر بررسی کنید:
```

Comment:
```suggestion
برای اطمینان، [مثال قبلی](codepen://rendering-elements/update-rendered-element) را با استفاده از ابزارهای مرورگر بررسی کنید:
```

Reply:
@sorousht I think it's better to be as close as possible to the original reference as long as it doesn't hurt the meaning and intended abstract in the target language.

I understand you wanna help readers about the specific tool in their browser and I fully support it.

But I think you should open a PR in the upstream (which is [reactjs/reactjs.org](https://github.com/reactjs/reactjs.org)) and make your defend there and if it gets accepted there we in the downstream will follow suit.
@tesseralis What do you think?

Reply:
Agreed. Translations should match the original as much as possible (e.g. don't add a new paragraph of content). A lot of translators have already made issues/PRs to the upstream repo -- you can too!

Reply:
@sJJdGG Thank you very much for looking over it closely. Your suggested change has been accepted.